[Kenneth Teo] iP

KennethTeo2002 opened the pull request:
## Nerf chatbot
> "Productivity is never an accident. It is always the result of a commitment to excellence, intelligent planning, and focused effort." - Paul J. Meyer [(source)](https://www.brainyquote.com/quotes/paul_j_meyer_393225)

Nerf is a state-of-the-arts task tracking application. It's
- text-based
- easy to use
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/KennethTeo2002/ip/releases/tag/A-Jar).
2. run it.
3. and you are good to go!😎

Best of all, it's **FREE!!**

Features include:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)

If you are a programmer and wish to view the code, feel free to clone it and modify to your liking.
The project uses gradle, thus to build the application, here's the command:
``` 
# git clone https://github.com/KennethTeo2002/ip.git
# cd ip
# gradlew build 
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

KennethTeo2002 committed:
Rename chatbot to Nerf

KennethTeo2002 committed:
Implement initial greetings and exit

KennethTeo2002 committed:
Implement echo

KennethTeo2002 committed:
Implement Add functionality

KennethTeo2002 committed:
Implement list functionality

KennethTeo2002 committed:
Create Task class

KennethTeo2002 committed:
Implement Mark and Unmark functionality

KennethTeo2002 committed:
Create subclasses of Task

KennethTeo2002 committed:
Implement subclasses into program

KennethTeo2002 committed:
Add automated test

KennethTeo2002 committed:
Implement error handling

KennethTeo2002 committed:
Fix testcases

KennethTeo2002 committed:
Add delete functionality

KennethTeo2002 committed:
Add overloaded constructor to Task classes

KennethTeo2002 committed:
Implement loading of save file

KennethTeo2002 committed:
Implement saving to file after task change

KennethTeo2002 committed:
Clean up debug statements

KennethTeo2002 committed:
Fix printing to fit test

KennethTeo2002 committed:
Merge branch 'branch-Level-7'

KennethTeo2002 committed:
Modify storage of datetime variable

KennethTeo2002 committed:
Update testcase to fit datetime input

KennethTeo2002 committed:
Merge branch 'branch-Level-8'

KennethTeo2002 committed:
Refactor code to extract more classes

KennethTeo2002 committed:
Add test/data to gitignore

KennethTeo2002 committed:
Merge branch 'branch-MoreOOP'

KennethTeo2002 committed:
Divide class into respective packages

KennethTeo2002 committed:
Merge branch 'A-Packages'

KennethTeo2002 committed:
Merge branch 'add-gradle-support'

KennethTeo2002 committed:
Update gradle configuration

KennethTeo2002 committed:
Add deadline task test cases

KennethTeo2002 committed:
Add event task test cases

KennethTeo2002 committed:
Merge branch 'branch-A-JUnit'

KennethTeo2002 committed:
Add header comments to classes and methods

KennethTeo2002 committed:
Tweak code to comply with coding standard

KennethTeo2002 committed:
Implement find functionality

KennethTeo2002 committed:
Merge branch 'branch-A-JavaDoc'

KennethTeo2002 committed:
Merge branch 'branch-A-CodingStandard'

KennethTeo2002 committed:
Merge branch 'branch-Level-9'

KennethTeo2002 committed:
Add template GUI from tutorial

KennethTeo2002 committed:
Tweak GUI styling

KennethTeo2002 committed:
Refactor chatbot to handle GUI input

KennethTeo2002 committed:
Refactor GUI to use fxml

KennethTeo2002 committed:
Move header comment to below class

KennethTeo2002 committed:
Merge branch 'branch-Level-10'

wuzengfu commented on src/main/java/nerf/task/Deadlines.java:
```diff
@@ -0,0 +1,49 @@
+
+    
+    /**
+     * Return format for saving to file.
```
Violation of coding standard on comments. `Return` should be `Returns`

wuzengfu commented on src/main/java/nerf/task/Task.java:
```diff
@@ -0,0 +1,66 @@
+    public Task(String description) {
+        this.description = description;
+    }
+    public Task(String description,boolean status) {
```
Violation of coding standard. There should be a white space behind the comma.
```java
public Task(String description, boolean status) {
```

wuzengfu commented on src/main/java/nerf/task/Task.java:
```diff
@@ -0,0 +1,66 @@
+     * @param keyword to check.
+     * @return true if keyword in description.
+     */
+    public boolean matchSearch(String keyword) {
```
Violation of coding quality/standard on naming. The method name should be something like `isKeywordMatched`

wuzengfu commented on src/main/java/nerf/task/TaskList.java:
```diff
@@ -0,0 +1,246 @@
+    /**
+     * Default constructor if save file is unable to read.
+     */
+
```
Violation of coding standard on JavaDoc. There should not be any line break between JavaDoc and method signature

wuzengfu commented on src/main/java/nerf/task/TaskList.java:
```diff
@@ -0,0 +1,246 @@
+    }
+
+    /**
+     * Add new task to list.
```
Violation of coding standards on JavaDoc. It should be `Adds new task to list.`. This applies to the rest of JavaDoc

wuzengfu reviewed (commented):
LGTM. Good structure of the program! I have noted some minor issues on coding standard/quality.

curtischang2510 commented on src/main/java/nerf/storage/Storage.java:
```diff
@@ -0,0 +1,82 @@
+     * 
+     * @param taskList list of task in string format.
+     */
+    public void save(List<String> taskList){
```
Coding standard violation: A space should be added before open curly bracket

curtischang2510 commented on src/main/java/nerf/task/Events.java:
```diff
@@ -0,0 +1,55 @@
+     * @return string format of task.
+     */
+    @Override
+    public String getSaveFormat(){
```
Coding standard violation: A space should be added before open curly bracket :)

curtischang2510 commented on src/main/java/nerf/task/ToDos.java:
```diff
@@ -0,0 +1,33 @@
+    public ToDos(String description) {
+        super(description);
+    }
+    public ToDos(String description, boolean status) {
```
Coding standard violation: boolean variables should sound like booleans. Perhaps something like 

`boolean isDone` or `isComplete`

curtischang2510 commented on src/main/java/nerf/task/TaskList.java:
```diff
@@ -0,0 +1,246 @@
+    }
+
+    /**
+     * Filter and find task based on keyword.
```
Coding standard violation: I believe `Filters` should be used instead of `Filter` for JavaDocs

curtischang2510 reviewed (commented):
Good job with the IP. I think your code is well written with really good structure. Code is generally very clean

KennethTeo2002 committed:
Add greeting banner to GUI

KennethTeo2002 committed:
Fix coding standard issues

KennethTeo2002 committed:
Add assert to some assumptions made

There are several methods that made assumptions about its input parameters.

Adding assert can validate these assumptions in case of invalid input.

Let's add some assert tests to ensure that the variables match the assumed values.

KennethTeo2002 committed:
Merge pull request #1 from KennethTeo2002/branch-A-Assertions

Add assert to some assumptions made

KennethTeo2002 committed:
Refactor code quality

There are several methods that have multiple levels of abstraction.

Implementing SLAP makes the code more readable with higher-levle logic.

Let's move some function logic to helper functions.

KennethTeo2002 committed:
Make happy path clearer

KennethTeo2002 committed:
Merge branch 'master' into branch-A-CodeQuality

KennethTeo2002 committed:
Merge pull request #2 from KennethTeo2002/branch-A-CodeQuality

Refactor code for better code quality

KennethTeo2002 committed:
Add setting of priority to task functionality

KennethTeo2002 committed:
Merge pull request #3 from KennethTeo2002/branch-C-Priority

Add setting of priority to task functionality

KennethTeo2002 committed:
Save priority to data file

KennethTeo2002 committed:
Update testcase to fit priority requirement

KennethTeo2002 committed:
Merge pull request #4 from KennethTeo2002/branch-C-Priority

Branch c priority

KennethTeo2002 committed:
Fix method naming

KennethTeo2002 committed:
Merge branch 'branch-refactor'

KennethTeo2002 committed:
Add user guide template

KennethTeo2002 committed:
Fix bug with invalid data in save file

KennethTeo2002 committed:
Update user guide to fit feature list

KennethTeo2002 committed:
Fix markdown rendering

KennethTeo2002 committed:
Fix user guide title

KennethTeo2002 committed:
Merge pull request #6 from KennethTeo2002/branch-A-UserGuide

Branch a user guide

KennethTeo2002 committed:
Fix header comment
